File: dotnet/Allmark/Render/ParagraphRenderer.cs
namespace Allmark.Render;

using Allmark.Types;

public static class ParagraphRenderer
{
    public static OutputRenderer Create()
    {
        return new OutputRenderer
        {
            Name = "paragraph",
            Render = Render,
        };
    }

    public static void Render(MarkdownNode node, RendererState state, bool? decode = true)
    {
        RenderUtils.StartNewLine(node, state);
        state.Output.Append("<p>");
        RenderUtils.InnerNewLine(node, state);
        RenderChildren.Execute(node, state);
        state.Output.Append("</p>");
        RenderUtils.EndNewLine(node, state);
    }
}

File: web/src/render-console/paragraphRenderer.ts
import type ConsoleRendererState from "../types/ConsoleRendererState";
import type MarkdownNode from "../types/MarkdownNode";
import type Renderer from "../types/Renderer";
import type RendererState from "../types/RendererState";
import renderChildren from "./renderChildren";

const renderer: Renderer = {
	name: "paragraph",
	render,
};
export default renderer;

function render(node: MarkdownNode, state: RendererState): void {
	const s = state as ConsoleRendererState;
	renderChildren(node, state);
	s.output += "\n\n";
}
